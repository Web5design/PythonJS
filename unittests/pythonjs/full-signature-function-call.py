def func(a, b, c=3, d=4, *args, **kwargs):
    print a, b, c, d, len(args), len(kwargs)
    return a+b+c+d

print '>>> func(1,2)'
print func(1,2)
print '>>> func(11,22,33,44,55,66)'
print func(11,22,33,44,55,66)
print '>>> func(1, 2, e=True)'
print func(1, 2, e=True)
print '>>> func(11,22,33,44,55,66,e=True,f=True)'
print func(11,22,33,44,55,66,e=True,f=True)


def func(a=1):
    print a

print '>>> func()'
func()
print '>>> func()'
func(2)
