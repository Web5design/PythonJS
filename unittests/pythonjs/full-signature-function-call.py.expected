>>> func(1,2)
1 2 3 4 0 0
10
>>> func(11,22,33,44,55,66)
11 22 33 44 2 0
110
>>> func(1, 2, e=True)
1 2 3 4 0 1
10
>>> func(11,22,33,44,55,66,e=True,f=True)
11 22 33 44 2 2
110
>>> func()
1
>>> func()
2
